src/main/kotlin/eu/europa/ec/eudi/sdjwt/SdJwtPresent.kt

```diff
@@ -89,24 +88,27 @@ fun <JWT> SdJwt.Issuance<JWT>.present(
 }
 
 /**
- *  Tries to create a presentation that discloses the claims are in [query]
- *  @param query a set of [JsonPointer] relative to the unprotected JSON (not the JWT payload). Pointers for
- *  * claims that are always disclosable can be omitted
- *  @receiver The issuance SD-JWT upon which the presentation will be based
- *  @return the presentation if possible to satisfy the [query]
- */
-fun SdJwt.Issuance<JwtAndClaims>.present(
-    query: Set<JsonPointer>,
-): SdJwt.Presentation<JwtAndClaims>? = present(query) { (_, claims) -> claims }
-
-/**
- * Tries to create a presentation that discloses the claims that satisfy
- * [query]
- * @param query a predicate for the claims to include in the presentation. The [JsonPointer]
- * is relative to the unprotected JSON (not the JWT payload)
+ * Tries to create a presentation that discloses the [requested claims][query].
+ * @param query a set of [ClaimPaths][ClaimPath] to include in the presentation. The [ClaimPaths][ClaimPath]
+ * are relative to the unprotected JSON (not the JWT payload)
+ * @param claimsOf a function to obtain the [Claims] of the [SdJwt.jwt]
  * @receiver The issuance SD-JWT upon which the presentation will be based
+ * @param JWT the type representing the JWT part of the SD-JWT
  * @return the presentation if possible to satisfy the [query]
  */
-fun SdJwt.Issuance<JwtAndClaims>.present(
-    query: (JsonPointer) -> Boolean,
-): SdJwt.Presentation<JwtAndClaims>? = present(query) { (_, claims) -> claims }
+fun <JWT> SdJwt.Issuance<JWT>.present(
```

Comment:
@babisRoutis Would it make sense to keep two present methods? One where the query is a set of ClaimPaths (as is), and an analogous one where the query is a set of JsonPointers instead of a JsonPointer predicate?

Reply:
Yes